app/src/main/java/com/matthaigh27/chatgptwrapper/activites/HomeActivity.kt

```diff
@@ -1,21 +1,91 @@
 package com.matthaigh27.chatgptwrapper.activites
 
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.pm.PackageManager
+import android.os.Build
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.tabs.TabLayout.TabGravity
 import com.matthaigh27.chatgptwrapper.R
 import com.matthaigh27.chatgptwrapper.fragments.ChatFragment
+import java.io.File
 
 
 class HomeActivity : AppCompatActivity() {
 
+    private val PERMISSIONS_REQUEST_CODE = 1
+
+    private val PERMISSIONS = arrayOf(
+        Manifest.permission.SEND_SMS,
+        Manifest.permission.READ_CONTACTS,
+        Manifest.permission.CALL_PHONE,
+        Manifest.permission.READ_EXTERNAL_STORAGE
+    )
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_home)
 
+        requestPermission()
+    }
+
+    private fun requestPermission() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            val notGrantedPermissions = PERMISSIONS.filter {
+                checkSelfPermission(it) != PackageManager.PERMISSION_GRANTED
+            }
+
+            if (notGrantedPermissions.isNotEmpty()) {
+                if (shouldShowRequestPermissionRationale(notGrantedPermissions[0])) {
+                    // show custom permission rationale
+                    AlertDialog.Builder(this)
```

Comment:
Let's create a function, which is responsible for managing all dialog in our app.